apps/base-docs/docs/pages/identity/smart-wallet/guides/linked-accounts/using-sub-accounts.mdx

```diff
@@ -0,0 +1,241 @@
+
+  const signMessage = useCallback(async () => {
+    if (!subAccountWalletClient || !subAccount) {
+      throw new Error('Subaccount wallet client or subaccount not found');
+    }
+
+    const message = 'Hello, world!';
+    const signature = await subAccountWalletClient.signMessage({
+      message,
+      account: subAccount,
+    });
+
+    setSignature(signature);
+    return signature;
+  }, [subAccountWalletClient, subAccount]);
+
+  // ... existing render logic
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', gap: '1rem' }}>
+      <p>Connected address: {address}</p>
+
+      {!subAccount && <button onClick={createSubAccount}>Create Sub Account</button>}
+
+      {subAccount && (
+        <div>
+          <p>Sub Account address: {subAccount}</p>
+          <button onClick={signMessage}>Sign Message</button>
+          {signature && <div style={{ overflowWrap: 'break-word' }}>Signature: {signature}</div>}
+        </div>
+      )}
+
+      <button onClick={disconnect}>Disconnect Wallet</button>
+    </div>
+  );
+}
+```
+
+## Sending Transactions
+
+Now let's implement transaction sending with gas sponsorship. To get a paymaster URL, sign up on [Coinbase Developer Platform](https://www.coinbase.com/developer-platform/products/paymaster) to get access
+to the free Base Sepolia testnet paymaster.
+
+Add this to your page.tsx:
+
+```tsx
+// app/page.tsx
+export default function Home() {
+  // ... existing state and hooks
+  const [txHash, setTxHash] = useState<string | null>(null);
+
+  const sendTransaction = useCallback(async () => {
+    if (!provider) {
+      throw new Error('Provider not found');
+    }
+
+    const txHash = await provider.request({
+      method: 'wallet_sendCalls',
+      params: [
+        {
+          chainId: baseSepolia.id,
+          calls: [
+            {
+              to: subAccount,
+              data: '0x',
+              value: 0,
+            },
+          ],
+          from: subAccount,
+          version: '1',
+          capabilities: {
+            paymasterService: {
+              url: 'YOUR_PAYMASTER_URL',
+            },
+          },
+        },
+      ],
+    });
+
+    setTxHash(txHash as string);
+    return txHash;
+  }, [provider, subAccount]);
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', gap: '1rem' }}>
+      {/* ... existing UI */}
+      {subAccount && (
+        <div>
+          {/* ... existing sub account UI */}
+          <button onClick={sendTransaction}>Send Transaction</button>
+          {txHash && (
+            <div style={{ overflowWrap: 'break-word' }}>
+              Transaction Hash:{' '}
+              <a
+                href={`https://sepolia.basescan.org/tx/${txHash}`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {txHash}
+              </a>
+            </div>
+          )}
+        </div>
+      )}
+    </div>
+  );
+}
+```
```

Comment:
I had to explicitly add my subaccount address to the allowlist configured on my paymaster in CDP portal for the "Send Transaction" button to work. I couldn't find  a way to bypass the need for allowlist in my config. Wondering if that step or note needs to go in this guide.